src/test/optimizer/strategy/subquery_to_join_reformulation_rule_test.cpp

```diff
@@ -0,0 +1,284 @@
+  // clang-format on
+  const auto actual_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(actual_lqp, expected_lqp);
+}
+
+TEST_F(SubqueryToJoinReformulationRuleTest, UncorrelatedNotInToAntiJoin) {
+  // SELECT * FROM a WHERE a.a NOT IN (SELECT b.a FROM b)
+  // clang-format off
+  const auto subquery_lqp =
+      ProjectionNode::make(expression_vector(node_table_b_col_a), node_table_b);
+
+  const auto subquery = lqp_subquery_(subquery_lqp);
+
+  const auto input_lqp =
+      PredicateNode::make(not_in_(node_table_a_col_a, subquery),
+                          node_table_a);
+
+  const auto expected_lqp =
+      JoinNode::make(JoinMode::Anti, equals_(node_table_a_col_a, node_table_b_col_a),
+                     node_table_a,
+                     ProjectionNode::make(expression_vector(node_table_b_col_a), node_table_b));
+  // clang-format on
+  const auto actual_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(actual_lqp, expected_lqp);
+}
+
+TEST_F(SubqueryToJoinReformulationRuleTest, SimpleCorrelatedInToInnerJoin) {
+  // SELECT * FROM a WHERE a.a IN (SELECT b.a FROM b WHERE b.b = a.b)
+  const auto parameter = correlated_parameter_(ParameterID{0}, node_table_a_col_b);
+
+  // clang-format off
+  const auto subquery_lqp =
+      ProjectionNode::make(expression_vector(node_table_b_col_a),
+          PredicateNode::make(equals_(node_table_b_col_b, parameter),
+              node_table_b));
+
+  const auto subquery = lqp_subquery_(subquery_lqp, std::make_pair(ParameterID{0}, node_table_a_col_b));
+
+  const auto input_lqp =
+      PredicateNode::make(in_(node_table_a_col_a, subquery),
+                          node_table_a);
+
+  const auto expected_lqp =
+      AggregateNode::make(expression_vector(node_table_a_col_a, node_table_a_col_b), expression_vector(),
+                          ProjectionNode::make(expression_vector(node_table_a_col_a, node_table_a_col_b),
+                                               PredicateNode::make(equals_(node_table_b_col_b, node_table_a_col_b),
+                                                                   JoinNode::make(JoinMode::Inner,
+                                                                                  equals_(node_table_a_col_a,
+                                                                                          node_table_b_col_a),
+                                                                                  node_table_a,
+                                                                                  node_table_b))));
+  // clang-format on
+  const auto actual_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(actual_lqp, expected_lqp);
+}
+
+// We currently do not support this reformulation, because an anti join would not preserve the columns from the right sub-tree.
+TEST_F(SubqueryToJoinReformulationRuleTest, ShouldNotReformulateSimpleCorrelatedNotInWithEqualityPredicate) {
```

Comment:
If this test doesn't work yet you could also just skip it (via `GTEST_SKIP();`) and don't have commented out code.